core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -916,8 +916,15 @@ public VirtualHostBuilder verboseResponses(boolean verboseResponses) {
      * @param shutdownOnStop whether to shut down the {@link AccessLogWriter} when the {@link Server} stops
      */
     public VirtualHostBuilder accessLogWriter(AccessLogWriter accessLogWriter, boolean shutdownOnStop) {
-        this.accessLogWriter = requireNonNull(accessLogWriter, "accessLogWriter");
-        shutdownAccessLogWriterOnStop = shutdownOnStop;
+        requireNonNull(accessLogWriter, "accessLogWriter");
+        if (this.accessLogWriter != null) {
```

Comment:
What do you think of also modifying the javadocs to reflect this behavioral change? Something along the lines of...

```
    /**
     * Adds an access log writer to this {@link VirtualHost}. If not set, the {@link AccessLogWriter} set via
     * {@link ServerBuilder#accessLogWriter(AccessLogWriter, boolean)} is used.
     * Note that even if an {@link AccessLogWriter} with a {@code false} {@code shutdownOnStop} value has been added,
     * all subsequently added {@link AccessLogWriter}s will be shut down irregardless of the specified {@code shutdownOnStop}.
     */
```

I think it would also be nice to update `ServerBuilder#accessLogWriter` as well.

Reply:
On second thought, I find this behavior odd.

```
Note that even if an {@link AccessLogWriter} with a {@code false} {@code shutdownOnStop} value has been added,
all subsequently added {@link AccessLogWriter}s will be shut down irregardless of the specified {@code shutdownOnStop}.
```

What do you think of either adding another default method to `AccessLogWriter` which also accepts a parameter on whether to chain shutdowns, and use it in this method?

https://github.com/line/armeria/blob/1008ba517cf7dd9df5699d0ca62e3f451464d66c/core/src/main/java/com/linecorp/armeria/server/logging/AccessLogWriter.java#L92

```
default AccessLogWriter andThen(AccessLogWriter after, boolean chainShutdown) {
...
if (chainShutdown) {
    f2 = after.shutdown();
} else {
    f2 = UnmodifiableFuture.completedFuture(null);
}
```

Reply:
> Note that even if an {@link AccessLogWriter} with a {@code false} {@code shutdownOnStop} value has been added,
all subsequently added {@link AccessLogWriter}s will be shut down irregardless of the specified {@code shutdownOnStop}.

Hm, I don't think it doesn't work in that way.
The `AccessLogWriter`s that are set with `true` will be shut down.

Reply:
Sorry, I had a type in my comment - I meant:

> Note that if an {@link AccessLogWriter} with a {@code true} {@code shutdownOnStop} value has been added,
all subsequently added {@link AccessLogWriter}s will be shut down irregardless of the specified {@code shutdownOnStop}